File: leetcode_by_python/3174.clear-digits.py
#
# @lc app=leetcode.cn id=3174 lang=python3
# @lcpr version=30204
#
# [3174] 清除数字
#


# @lcpr-template-start

# @lcpr-template-end
# @lc code=start
class Solution:
    def clearDigits(self, s: str) -> str:
        res = []
        for c in s:
            if c.isdigit():
                res.pop()
            else:
                res.append(c)
        return ''.join(res)
# @lc code=end



#
# @lcpr case=start
# "abc"\n
# @lcpr case=end

# @lcpr case=start
# "cb34"\n
# @lcpr case=end

#

File: leetcode_by_rust/src/bin/3174.clear-digits.rs
/*
 * @lc app=leetcode.cn id=3174 lang=rust
 * @lcpr version=30204
 *
 * [3174] 清除数字
 */


// @lcpr-template-start

// @lcpr-template-end
// @lc code=start
impl Solution {
    pub fn clear_digits(s: String) -> String {
        let mut stack = Vec::new();
        for c in s.chars() {
            match c {
                '0'..='9' => {
                    stack.pop();
                }
                _ => {
                    stack.push(c);
                }
            }
        }
        stack.iter().collect()
    }
}
// @lc code=end



/*
// @lcpr case=start
// "abc"\n
// @lcpr case=end

// @lcpr case=start
// "cb34"\n
// @lcpr case=end

 */

